### DISCUSSION TITLE
mod-ah-bot to work in container?

### DISCUSSION THREAD

**Discussion by Balubablahblah** [azerothcore/azerothcore-wotlk#9384] [Modules]
Hello!

Got a wonderful container up and running very smoothly, thank you Azerothcore dev team! Up and running for the simple enjoyment for a few friends, the container setup replacing a poor force built and hack'n'slashed old manual build of a alternate distribution.

One request that popped up quickly was the possibilty to add the auction house. Instructions to add modules seemed pretty forward, so:
1) Downloaded and extracted the module directory, so ./azerothcore-wotlk/modules/mod-ah-bot-master/
2) Tables and content created (mod_auctionhousebot.sql) in the world DB
3) Files in module ./conf/ put and duplicated without the .dist under ./azerothcore-wotlk/env/docker/etc/modules/
4) Separate account created, character created, information put in the mod_ahbot.conf file
| guid | account | name     |
+------+---------+----------+
|    7 |      15 | Ebenezer |

5) ./acore.sh docker build

But sadly no items in the auction house. Destroyed the containers and tried a complete rebuild, but no luck there either, but it ALMOST gets me there, ending with:

...
[100%] Linking CXX executable worldserver
/usr/bin/ld: ../scripts/libscripts.a(ScriptLoader.cpp.o): in function `AddScripts()':
/azerothcore/var/build/obj/src/server/scripts/gen_scriptloader/static/ScriptLoader.cpp:83: undefined reference to `Addmod_ah_bot_masterScripts                                                              ()'
clang: error: linker command failed with exit code 1 (use -v to see invocation)
make[2]: *** [src/server/worldserver/CMakeFiles/worldserver.dir/build.make:174: src/server/worldserver/worldserver] Error 1
make[1]: *** [CMakeFiles/Makefile2:1159: src/server/worldserver/CMakeFiles/worldserver.dir/all] Error 2

Enrolled a friend who is much more skilled in both containers and coding to see if he could get it rolling, but he gave up after finding the current module throughout the code using 32 bit objects, and not 64 bit objects as the main Azerothcore code...?

Here's hoping someone here have got the auction house working in a container installation, and can help and point me (and others?) in the right direction on what needs to done here. :)

**Comment by Winfidonarleyan** [comment]
Please use git glone

**Reply by MrH0042** [reply]
Hi,

(yes, both repos cloned/pulled to latest&greatest)

more specifically, the mod-ahbot C++ code uses things like SQLTransaction foo where the main AZCore code apparently have moved over to e.g. CharacterDatabaseTranslation foo and WorldDatabaseTranslation xyz. Also, the item->GetGUIDLow() method no longer exists, meaning the part of the diff that says e.g:

...
ahbot.patch:+        if (owner && owner->GetGUIDLow() != auctionbot->GetAHBplayerGUID())
....
ahbot.patch:+            oldBidder->GetSession()->SendAuctionBidderNotification(auction->GetHouseId(), auction->Id, auctionbot->GetAHBplayerGUID(), newP auction->GetAuctionOutBid(), auction->item_template);

fails.  The second line fails as the SendAuctionBidderNotification method now expects a ObjectGuid object here instead of the uint32 returned by the GetAHBplayerGUID().

Looking at ahbot.patch, line numbers of the hunks etc, it looks like src/server/game/AuctionHouse/AuctionHouseMgr.cpp have seen fairly large updates since the diff was created.

Someone more familiar with the code is needed to adopt to these changes. I gave it a couple of hours but had to give up, AZCore being entirely new to me.

Cheers,
  H.
